WalletWasabi/WebClients/WasabiNostrClient.cs

```diff
@@ -0,0 +1,79 @@
+using NNostr.Client;
+using NNostr.Client.Protocols;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.WebClients;
+
+public class WasabiNostrClient(INostrClient nostrClient)
+{
+	private readonly string _nostrSubscriptionID = Guid.NewGuid().ToString();
+	private readonly Channel<ReleaseInfo> _updateChannel = Channel.CreateUnbounded<ReleaseInfo>();
+	private readonly Dictionary<string, NostrEvent> _events = new();
+
+	public ChannelReader<ReleaseInfo> EventsReader => _updateChannel.Reader;
+
+	public async Task ConnectAnsSubscribeAsync(CancellationToken cancel)
+	{
+		nostrClient.EventsReceived += OnNostrEventsReceived;
+		nostrClient.EoseReceived += OnEndOfStoredEvents;
+
+		await nostrClient.ConnectAndWaitUntilConnected(cancel).ConfigureAwait(false);
+
+		var nostrSubscriptionFilter = new NostrSubscriptionFilter {
+			Kinds = [1],
+			Authors = [NIP19.FromNIP19Npub(Constants.WasabiTeamNostrPubKey).ToHex()],
+			Limit = 1};
+		await nostrClient.CreateSubscription(_nostrSubscriptionID, [nostrSubscriptionFilter], cancel).ConfigureAwait(false);
+	}
+
+	private void OnEndOfStoredEvents(object? sender, string e)
+	{
+		nostrClient.EventsReceived -= OnNostrEventsReceived;
+		_updateChannel.Writer.TryComplete();
+	}
```

Comment:
This event handler is called multiple times, and so far from my testing it looks like we call `Writer.TryComplete` before we process the events (before we use `TryWrite` in the EventReceived event handler)
Therefore when we read the Channel, it's empty.
It doesn't happen every time, but more often than not. Can it be a race condition?

Reply:
With `Logger.LogInfo` it looks like this
```
2025-04-07 17:16:49.922 [15] INFO       WasabiNostrClient.OnEndOfStoredEvents (41)      Completed UpdateChannel Writer.
2025-04-07 17:16:49.923 [16] INFO       UpdateManager.ProcessReleaseEventsAsync (63)    Read count from UpdateChannel: 0
2025-04-07 17:16:49.926 [14] INFO       WasabiNostrClient.OnNostrEventsReceived (76)    Writing releaseInfo to channel: 2.6.0
2025-04-07 17:16:49.928 [14] INFO       WasabiNostrClient.OnEndOfStoredEvents (45)      Failed to complete UpdateChannel Writer, probably already completed.
```

Reply:
Done